src/core/coap/coap.cpp

```diff
@@ -1695,6 +1702,23 @@ const otCoapTxParameters TxParameters::kDefaultTxParameters = {
     kDefaultMaxRetransmit,
 };
 
+//----------------------------------------------------------------------------------------------------------------------
+
+Resource::Resource(const char *aUriPath, RequestHandler aHandler, void *aContext)
+{
+    mUriPath = aUriPath;
+    mHandler = aHandler;
+    mContext = aContext;
+    mNext    = nullptr;
```

Comment:
nit: use initializer list?

Reply:
I don't think we can use initializer list here since these fields are not defined by `Resource` class itself and are instead inherited from the base class `otCoapResource`.